server/src/main/java/com/flipkart/varadhi/services/DefaultAuthZService.java

```diff
@@ -0,0 +1,119 @@
+package com.flipkart.varadhi.services;
+
+import com.flipkart.varadhi.auth.RoleBindingNode;
+import com.flipkart.varadhi.entities.ResourceType;
+import com.flipkart.varadhi.entities.RoleAssignmentUpdate;
+import com.flipkart.varadhi.exceptions.InvalidOperationForResourceException;
+import com.flipkart.varadhi.exceptions.ResourceNotFoundException;
+import com.flipkart.varadhi.spi.db.MetaStore;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class DefaultAuthZService {
+    private final MetaStore metaStore;
+
+    public DefaultAuthZService(MetaStore metaStore) {
+        this.metaStore = metaStore;
+    }
+
+    public RoleBindingNode createRoleBindingNode(String resourceId, ResourceType resourceType) {
+        if (!isResourceValid(resourceId, resourceType)) {
+            throw new InvalidOperationForResourceException(String.format(
+                    "Invalid resource id(%s) for resource type(%s).",
+                    resourceId,
+                    resourceType
+            ));
+        }
+        RoleBindingNode node = new RoleBindingNode(resourceId, resourceType, new HashMap<>(), 0);
+        metaStore.createRoleBindingNode(node);
+        return node;
+    }
+
+    public List<RoleBindingNode> getAllRoleBindingNodes() {
```

Comment:
Why is this needed ? I mean any use cases which needs all role binding nodes ?

Reply:
this is mainly for debug purposes